### PR TITLE
Week1-Browser-Osama Bazbooz

### PR DESCRIPTION
**Pull request by obazbooz**

### CODE REVIEW

File: 2-Browsers/Week1/homework/ex1-bookList/index.js [hunk 1]
```diff
@@ -39,7 +39,34 @@ const myBooks = [
...
-  // TODO your code goes in here, return the ul element
+  const ulEl = document.createElement('ul');
+  books.forEach((book, index) => {
+    const liEl = document.createElement('li');
+    const pEl = document.createElement('p');
+    const imgEl = document.createElement('img');
+    pEl.textContent = `${book.title} - ${book.author}`;
+    if (index === 0) {
+      imgEl.src = 'assets/the_design_of_everyday_things.jpg';
+      imgEl.alt = 'Image Not Found!';
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex1-bookList/index.js:50]: This line is being used regardless of the value of `index`... it probably doesn't belong in this `if-statement` then

File: test-runner.log [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+2021-11-02 11:16:17 error: *** Spell Checker Report ***
+
+
+2021-11-02 11:16:44 info: ---------------------------------------
+2021-11-02 11:16:44 info: >>> Running Unit Test `ex5-catWalk` <<<
+2021-11-02 11:16:44 info: ---------------------------------------
+2021-11-02 11:16:55 info: All unit tests passed.
+2021-11-02 11:16:57 error: *** Spell Checker Report ***
+
+
```
Comment (cgduncan7) [inline test-runner.log:36]: Woohoooooo! All tests passed! 💯 🍻

File: 2-Browsers/Week1/homework/ex1-bookList/index.js [hunk 3]
```diff
@@ -39,7 +39,34 @@ const myBooks = [
 ];
 
 function createBookList(books) {
-  // TODO your code goes in here, return the ul element
+  const ulEl = document.createElement('ul');
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex1-bookList/index.js:42]:
```suggestion
  const ulElement = document.createElement('ul');
```

Don't be afraid to have long variable names.. It's much more readable 😃

File: 2-Browsers/Week1/homework/ex2-aboutMe/index.js [hunk 4]
```diff
@@ -10,4 +10,23 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+let spanElHomeTown = document.getElementById('hometown');
+let pElName = document.createElement('p');
+let pElFood = document.createElement('p');
+let pElHome = document.createElement('p');
+pElName.textContent = 'Osama';
+pElFood.textContent = 'Sushi';
+pElHome.textContent = 'Netherlands';
+spanElNickname.replaceWith(pElName);
+spanElFavFood.replaceWith(pElFood);
+spanElHomeTown.replaceWith(pElHome);
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex2-aboutMe/index.js:28]: You don't need to literally replace the `span`. Could just change the text of the `span` but this works.

File: 2-Browsers/Week1/homework/ex2-aboutMe/index.js [hunk 5]
```diff
@@ -10,4 +10,23 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+pElName.style.display = 'inline';
+pElFood.style.display = 'inline';
+pElHome.style.display = 'inline';
+liEl.forEach((element) => element.classList.add('list-item'));
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex2-aboutMe/index.js:32]: This is arranged in a very confusing manner. Adding some line breaks between different logically grouped parts of your code along with comments will greatly improve readability

File: 2-Browsers/Week1/homework/ex2-aboutMe/index.js [hunk 6]
```diff
@@ -10,4 +10,23 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+let ulEl = document.querySelectorAll('ul');
+let liEl = document.querySelectorAll('li');
+let spanElNickname = document.getElementById('nickname');
+let spanElFavFood = document.getElementById('fav-food');
+let spanElHomeTown = document.getElementById('hometown');
+let pElName = document.createElement('p');
+let pElFood = document.createElement('p');
+let pElHome = document.createElement('p');
+pElName.textContent = 'Osama';
+pElFood.textContent = 'Sushi';
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex2-aboutMe/index.js:24]: 🍣 me too

File: 2-Browsers/Week1/homework/ex3-hijackLogo.js [hunk 7]
```diff
@@ -8,7 +8,11 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
    HackYourFuture logo instead.
 ------------------------------------------------------------------------------*/
 function hijackGoogleLogo() {
-  // TODO your code goes in here
+  const logoEl = document.getElementsByClassName('lnXdpd')[0];
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex3-hijackLogo.js:11]: 🕵️

File: 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js [hunk 8]
```diff
@@ -7,8 +7,18 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
   second). Use `setInterval()` to make sure the time stays current.
 2. Have the function execute when it's loading in the browser.
 ------------------------------------------------------------------------------*/
+
 function addCurrentTime() {
-  // TODO complete this function
+  let timeEl = new Date();
+  divEl.textContent = `The current time is: (H:${timeEl.getHours()} M:${timeEl.getMinutes()} S:${timeEl.getSeconds()} )`;
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js:13]: This isn't what was meant by `HH:MM:SS` format. Output the text like you would expect to see on a digital clock!
For example: 
> 12:34:56

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js [hunk 9]
```diff
@@ -22,7 +22,38 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
    https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif
 -----------------------------------------------------------------------------*/
 function catWalk() {
-  // TODO complete this function
-}
+  if (steps >= screen.width) steps = 0;
+  else if (steps === 400) {
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex5-catWalk/index.js:26]: What if my screen width is `100000`? `400` won't be halfway... How can you make sure that it always occurs at the halfway point?

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js [hunk 10]
```diff
@@ -22,7 +22,38 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+    steps += 10;
+  }
+
+  imgEl.style.left = steps + 'px';
 
-// TODO execute `catWalk` when the browser has completed loading the page
+  function letsDance() {
+    imgEl.src =
+      'https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif';
+  }
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex5-catWalk/index.js:45]: This function is unused

**Review by cgduncan7 — CHANGES_REQUESTED** [review]
Great work @obazbooz 👏 Everything is working but there are some issues with ex4 and ex5. Please take a look at my comments and see if you can address them 👍

**Comment by robvk** [comment]
@obazbooz , please have a look at the comments

**Pull request closed without merging**